tests/unit/ext/switches/brocade.py

```diff
@@ -0,0 +1,229 @@
+    <rspan-vlan y:self="/rest/config/running/interface/TenGigabitEthernet/%22104/0/10%22/switchport/trunk/allowed/rspan-vlan"/>
+    <vlan y:self="/rest/config/running/interface/TenGigabitEthernet/%22104/0/10%22/switchport/trunk/allowed/vlan">
+      <add>4001,4025</add>
+    </vlan>
+  </allowed>
+
+  <tag y:self="/rest/config/running/interface/TenGigabitEthernet/%22104/0/10%22/switchport/trunk/tag">
+    <native-vlan>true</native-vlan>
+  </tag>
+
+  <native-vlan>10</native-vlan>
+
+</trunk>
+"""
+
+TRUNK_PAYLOAD = '<mode><vlan-mode>trunk</vlan-mode></mode>'
+
+TRUNK_NATIVE_PAYLOAD = '<trunk><native-vlan>102</native-vlan></trunk>'
+
+TRUNK_VLAN_PAYLOAD = '<vlan><add>102</vlan></vlan>'
+
+TRUNK_REMOVE_VLAN_PAYLOAD = '<vlan><remove>102</remove></vlan>'
+
+INTERFACE1 = '104/0/10'
+INTERFACE2 = '104/0/18'
+INTERFACE3 = '104/0/20'
+
+
+class TestBrocade(object):
+    @pytest.fixture()
+    def switch(self):
+        return brocade.Brocade(
+            hostname='http://example.com',
+            username='admin',
+            password='admin',
+            interface_type='TenGigabitEthernet'
+        ).session()
+
+    def test_get_port_networks(self, switch):
+        with requests_mock.mock() as mock:
+            mock.get(switch._construct_url(INTERFACE1, suffix='trunk'),
+                     text=TRUNK_NATIVE_VLAN_RESPONSE_WITH_VLANS)
+            mock.get(switch._construct_url(INTERFACE2, suffix='trunk'),
+                     text=TRUNK_NATIVE_VLAN_RESPONSE_NO_VLANS)
+            mock.get(switch._construct_url(INTERFACE3, suffix='trunk'),
+                     text=TRUNK_VLAN_RESPONSE)
+            response = switch.get_port_networks([INTERFACE1,
+                                                 INTERFACE2,
+                                                 INTERFACE3])
+            assert response == {
+                INTERFACE1: [('vlan/native', '10'),
+                             ('vlan/4001', '4001'),
+                             ('vlan/4025', '4025')],
+                INTERFACE2: [('vlan/native', '10')],
+                INTERFACE3: [('vlan/1', '1'),
+                             ('vlan/4001', '4001'),
+                             ('vlan/4004', '4004'),
+                             ('vlan/4025', '4025'),
+                             ('vlan/4050', '4050')]
+            }
+
+    def test_get_mode(self, switch):
+        with requests_mock.mock() as mock:
+            mock.get(switch._construct_url(INTERFACE1, suffix='mode'),
+                     text=MODE_RESPONSE_ACCESS)
+            response = switch._get_mode(INTERFACE1)
+            assert response == 'access'
+
+            mock.get(switch._construct_url(INTERFACE1, suffix='mode'),
+                     text=MODE_RESPONSE_TRUNK)
+            response = switch._get_mode(INTERFACE1)
+            assert response == 'trunk'
+
+    def test_apply_networking(self, switch):
+        # Note(knikolla): Create a Nic connected to the Switch
```

Comment:
Basic docs describing what this does probably shouldn't have your name next to them; I tend to reserve that for stuff where my identity is actually relevant.